catalog/spatial/registry/registry-publication-update-handler/src/main/java/org/codice/ddf/registry/publication/RegistryPublicationHandler.java

```diff
@@ -41,14 +46,25 @@
 
     private static final int SHUTDOWN_TIMEOUT_SECONDS = 60;
 
+    private static final String KARAF_LOCAL_ROLE = "karaf.local.roles";
+
     private final RegistryPublicationService registryPublicationService;
 
     private final ExecutorService executor;
 
+    private javax.security.auth.Subject subject;
+
     public RegistryPublicationHandler(RegistryPublicationService registryPublicationService,
             ExecutorService service) {
         this.registryPublicationService = registryPublicationService;
         this.executor = service;
+
+        Set<Principal> principals = new HashSet<>();
```

Comment:
I am updating the Security class to support PrivilegedExceptionActions as well as just PrivilegedActions so i will be updating both of these to use the new Security method